tests/code/test_search.py

```diff
@@ -1,66 +1,77 @@
-#
-#@pytest.mark.integration
-#def test_noResults():
-#
-#    search = searchVolunteers("XXZ")
-#    assert search == '{}'
-#
 @pytest.mark.integration
-def test_searchStudents():
+def test_searchUsers():
 
     #tests that the search works
-
-    query = 'za'
-    test1 = User.select().where(User.firstName == query)
-    searchResults = searchUsers(query)
-    assert searchResults['neillz'] == test1
-
+    searchResults = searchUsers('za')
+    assert searchResults['neillz'] == model_to_dict(User.get_by_id('neillz'))
 
     #test for last name
-    query = 'khatt'
-    test2 = User.select().where(User.firstName == query)
-    searchResults = searchUsers(query)
-    assert searchResults['khatts'] == test2
+    searchResults = searchUsers('khat')
+    assert searchResults['khatts'] == model_to_dict(User.get_by_id('khatts'))
 
-    #test with non existing username
-    query = 'abc'
-    searchResults = searchUsers(query)
-    assert len(searchResults) == 0
+    #test for first and last name
+    searchResults = searchUsers('za nei')
+    assert searchResults['neillz'] == model_to_dict(User.get_by_id('neillz'))
 
-    query = 'its easy as'
-    searchResults = searchUsers(query)
+    #test with non existing username
+    searchResults = searchUsers('abc')
     assert len(searchResults) == 0
 
-    query = '123'
-    searchResults = searchUsers(query)
+    searchResults = searchUsers('its wrong')
     assert len(searchResults) == 0
 
     #test with multiple users matching query
     with mainDB.atomic() as transaction:
-        User.get_or_create(username = 'sawconc', firstName = 'Candace', lastName = 'Sawcon', bnumber = '021556782', isStudent = True)
-        query = 'sa'
-        searchResults = searchUsers(query)
-        test3 = User.select().where(User.firstName == query)
+        secondUser = User.create(username = 'sawconc', firstName = 'Candace', lastName = 'Sawcon', bnumber = '021556782', email = 'test@berea.edu', isStudent = True)
+
+        searchResults = searchUsers('sa')
         assert len(searchResults) == 2
-        assert searchResults['lamichhanes2'] == test3
-        assert 'Sawcon' in searchResults["sawconc"].values()
+        assert searchResults['lamichhanes2'] == model_to_dict(User.get_by_id('lamichhanes2'))
+        assert searchResults["sawconc"] == model_to_dict(secondUser)
         assert '(555)555-5555' in searchResults["lamichhanes2"].values()
+
         transaction.rollback()
+
+@pytest.mark.integration
+def test_searchUser_categories():
+
+    # tests that the search categories exclude properly
+    searchResults = searchUsers('za', 'instructor')
+    assert len(searchResults) == 0
+    searchResults = searchUsers('scott', 'admin')
+    assert len(searchResults) == 0
+    searchResults = searchUsers('sreyn', 'studentstaff')
+    assert len(searchResults) == 0
+
+    # tests that the search categories include properly
+    searchResults = searchUsers('sco', 'instructor') #faculty
```

Comment:
Will want a space to the right of the "#" for consistency  #faculty --> # faculty